### PR TITLE
[BUG] :bug: checking id match on PUT should be case insensitive (#657)

### PR DESCRIPTION
**Pull request by emmanuelbar**
Signed-off-by: Emmanuel Barrios <emmanuel.barrios@insee.fr>

### CODE REVIEW

File: sugoi-api-rest-commons-services/src/main/java/fr/insee/sugoi/commons/services/controller/technics/SugoiAdviceController.java [hunk 1]
```diff
@@ -14,33 +14,7 @@
...
-import fr.insee.sugoi.core.exceptions.RealmAlreadyExistException;
-import fr.insee.sugoi.core.exceptions.RealmNotCreatedException;
-import fr.insee.sugoi.core.exceptions.RealmNotFoundException;
-import fr.insee.sugoi.core.exceptions.StoragePolicyNotMetException;
-import fr.insee.sugoi.core.exceptions.UserAlreadyExistException;
-import fr.insee.sugoi.core.exceptions.UserNotCreatedException;
-import fr.insee.sugoi.core.exceptions.UserNotFoundByMailException;
-import fr.insee.sugoi.core.exceptions.UserNotFoundException;
-import fr.insee.sugoi.core.exceptions.UserStorageNotFoundException;
+import fr.insee.sugoi.core.exceptions.*;
```
Comment (CChemin) [inline sugoi-api-rest-commons-services/src/main/java/fr/insee/sugoi/commons/services/controller/technics/SugoiAdviceController.java:17]: can you use explicit import ?

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java [hunk 2]
```diff
@@ -233,8 +234,8 @@ public ResponseEntity<Application> updateApplication(
       @Parameter(description = "Application to update", required = true) @RequestBody
           Application application) {
 
-    if (!application.getName().equals(applicationName)) {
-      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    if (!application.getName().equalsIgnoreCase(applicationName)) {
+      throw new ApplicationNameNotMatchingException(applicationName, application.getName());
```
Comment (CChemin) [inline sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java:238]: why not use idnotmatchingexception as well ?

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java [hunk 3]
```diff
@@ -211,7 +212,7 @@ public ResponseEntity<Application> createApplication(
             content = {@Content(mediaType = "application/json")})
       })
   @PreAuthorize("@NewAuthorizeMethodDecider.isAppManager(#realm,#applicationName)")
-  public ResponseEntity<Application> updateApplication(
+  public ResponseEntity<?> updateApplication(
```
Comment (CChemin) [inline sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java:215]: is there a reason to remove this ?

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/OrganizationController.java [hunk 4]
```diff
@@ -309,8 +310,8 @@ public ResponseEntity<?> updateOrganizations(
       Authentication authentication,
       @Parameter(description = "Organization to update", required = false) @RequestBody
           Organization organization) {
-    if (!organization.getIdentifiant().equals(id)) {
-      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    if (!organization.getIdentifiant().equalsIgnoreCase(id)) {
```
Comment (CChemin) [inline sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/OrganizationController.java:313]: can you also manage the case when the object doesn't have its id ? (I think it's okay if it is dealt with the same Exception)

File: sugoi-api-rest-commons-services/src/main/java/fr/insee/sugoi/commons/services/controller/technics/SugoiAdviceController.java (hunk 1 above)
Comment (emmanuelbar) [reply sugoi-api-rest-commons-services/src/main/java/fr/insee/sugoi/commons/services/controller/technics/SugoiAdviceController.java:17]: My IDE won't let me 😢

**Review by CChemin — APPROVED** [review]

**Pull request merged**